gamemodes/terrortown/gamemode/client/cl_main.lua

```diff
@@ -729,16 +731,15 @@ net.Receive("TTT2PlayerAuthedShared", function(len)
 	local steamid64 = net.ReadString()
 	local name = net.ReadString()
 
+	-- checking for bots
+	if steamid64 == "" then
+		steamid64 = nil
+	end
+
 	hook.Run("TTT2PlayerAuthed", steamid64, name)
 end)
 
 hook.Add("TTT2PlayerAuthed", "TTT2CacheAvatar", function(steamid64, name)
```

Comment:
It would be cool, if you could also make this a named function and call it inside the function above instead of adding it to the hook :) (even if that does not have to be done in this PR)

Reply:
It can be useful globally, but where should i put it?

Reply:
I'm not that deep in this problem, but if it is something that can be put inside of `xxx.yyy`, then put it in `lua/ttt2/libraries`

Reply:
To summarize the problem:
https://wiki.facepunch.com/gmod/Player:SteamID64
→ SteamID64 returns different values on client and server for bots.
→ When the SteamID is sent via a netmessage the client gets confused easily.

It would either need 2 functions:
- A function for the client to set the received steamid to nil, when it's X.
- A function for the server to set a steamid to X when it's a bots before sending.

or one function:
- A function for the client which detects if it's a bots id and sets it to nil